tsml_estimator_evaluation/_wip/estimator_from_file/hivecote.py

```diff
@@ -87,23 +91,87 @@ def _fit(self, X, y):
             file_name = 'trainResample' + str(self.random_state) + '.csv'
 
         acc_list = []
+        all_lines = []
         for path in self.file_paths:
             f = open(path + file_name, "r")
             lines = f.readlines()
             line2 = lines[2].split(",")
 
             #   verify file matches data, i.e. n_instances and n_classes
             if len(lines)-3 != len(X): # verify n_instances
-                print("ERROR n_instances does not match in: ", path + file_name)
+                print("ERROR n_instances does not match in: ", path + file_name, len(lines) - 3, len(X))
             if len(np.unique(y)) != int(line2[5]): # verify n_classes
                 print("ERROR n_classes does not match in: ", path + file_name, len(np.unique(y)), line2[5])
 
             acc_list.append(float(line2[0]))
+            all_lines.append(lines)
+
+        if self.tune_alpha:
+            self.alpha = self._tune_alpha(all_lines)
 
         #   add a weight to the weight list based on the files accuracy
         for acc in acc_list:
             self._weights.append(acc ** self.alpha)
 
+    def _tune_alpha(self, all_files_lines):
+        alpha = 1
+        n_splits = 5
+        n_samples = len(all_files_lines[0])-3
+        X = np.zeros((n_samples, 4, self.n_classes_))
+        y = np.zeros(n_samples, dtype=int)
+        Acc = np.zeros((n_samples, 4))
+        for i, lines in enumerate(all_files_lines):
+            for j in range(n_samples):
+                line = lines[j+3].split(",")
+                acc_0 = float(line[3])
+                acc_1 = float(line[4])
+                X[j][i] = [acc_0, acc_1]
+                if line[0] == line[1]:
+                    if acc_0 > acc_1:
+                        Acc[j][i] = acc_0
+                    else:
+                        Acc[j][i] = acc_1
+                else:
+                    if acc_0 < acc_1:
+                        Acc[j][i] = acc_0
+                    else:
+                        Acc[j][i] = acc_1
+
+                y[j] = int(line[0]) # its getting y 4 times, not efficient
+
+        alpha_values = range(3000,4000) # 3032
+        avg_acc_alpha = np.zeros(len(alpha_values))
+        for i, alpha in enumerate(alpha_values):
+            kf = KFold(n_splits=n_splits)
+            # print(kf.get_n_splits(X))
+            # print(kf)
+            avg_acc = np.zeros(n_splits)
+            for j, (train_index, test_index) in enumerate(kf.split(X)):
+                print(f"Fold {j}:")
+                print(f"  Train: index={train_index}")
+                print(f"  Test:  index={test_index}")
+                x_test_set = X[test_index]
+                y_test_set = y[test_index]
+                acc_list = Acc[train_index].sum(axis=0)/len(train_index)
+                weight_list = acc_list ** alpha
```

Comment:
What we want to do is select all the train probabilities, turn them in to predictions i.e.
```
train_preds = self.classes_[np.argmax(X[train_index, n], axis=1)]
train_acc = accuracy_score(y, train_preds)
```
and then apply train_acc ** alpha to the test set. This should be done for each classifier/file